### PR TITLE
data preprocessing code added

### PR DESCRIPTION
**Pull request by leijerry888**
Add data preprocess step to fill / drop NaNs. two additional fields are needed in the case schema instance. So corresponding case generation script needs to be updated. See below for example of added elements under a verification case.
`            "data_processing":{
                "fillna": false,
                "then_dropna":true
            }
`

### CODE REVIEW

**Comment by leijerry888** [comment]
@lymereJ  for unit testing, I am thinking of making a small dataset sample for a valid verification case testing, for which using different fill/drop nan options causes different reported results. What do you think? Do you have some candidate dataset for making this test case?

**Comment by leijerry888** [comment]
This PR is for `animate_tspr`. If we don't plan to merge `animate_tspr` back to `master`, then we should still bring this PR back to `master`

**Comment by lymereJ** [comment]
> @lymereJ for unit testing, I am thinking of making a small dataset sample for a valid verification case testing, for which using different fill/drop nan options causes different reported results. What do you think? Do you have some candidate dataset for making this test case?

Yup, that sounds good, I can share some of the results we got from the last round of testing.

File: src/item.py [hunk 1]
```diff
@@ -182,9 +188,38 @@ def read_points_values(self, csv_path=None, idf_path=None, idd_path=None):
         """
         self.read_output_variables(csv_path)
         self.read_idf_obj_values(idf_path, idd_path)
+        self.df = DateTimeEP(self.df, self.DF_YEAR).transform()
+        self.data_processing(
+            fill_method=self.item["data_processing"]["fillna"],
+            then_dropna=self.item["data_processing"]["then_dropna"],
```
Comment (lymereJ) [inline src/item.py:194]: I'm not sure if about the `then_dropna` term. I really don't have a strong suggestion, the only thing I can think about is `drop_remaining_na` or `dropna_after_fillna`.

**Review by lymereJ — COMMENTED** [review]
Looks great! I'll approve now to not hold it up but we'll probably need a (passing) unit test before merging.

File: src/item.py (hunk 1 above)
Comment (yanchenpnnl) [reply src/item.py:194]: Maybe add a set (before and after) of data file for testing the code, and showing the before/after difference.

**Review by yanchenpnnl — APPROVED** [review]
Maybe add a set (before and after) of data file for testing the code, and showing the before/after difference.  I tentatively approve it for now.

**Pull request merged**